cmd/xl-v1-object.go

```diff
@@ -452,14 +453,35 @@ func (xl xlObjects) getObjectInfo(ctx context.Context, bucket, object string) (o
 	// Read metadata associated with the object from all disks.
 	metaArr, errs := readAllXLMetadata(ctx, xl.getDisks(), bucket, object)
 
-	// get Quorum for this object
+	// Get quorum for this object
 	readQuorum, _, err := objectQuorumFromMeta(ctx, xl, metaArr, errs)
 	if err != nil {
 		return objInfo, err
 	}
 
-	if reducedErr := reduceReadQuorumErrs(ctx, errs, objectOpIgnoredErrs, readQuorum); reducedErr != nil {
-		return objInfo, reducedErr
+	const xlCorruptedSuffix = ".CORRUPTED"
+
+	// Having read quorum means we have xl.json in at least N/2 disks.
+	if !strings.HasSuffix(object, xlCorruptedSuffix) {
+		// We can consider an object data not reliable
+		// when xl.json is not found in read quorum disks.
+		var notFoundXLJSON int
+		for _, readErr := range errs {
+			if readErr == errFileNotFound {
+				notFoundXLJSON++
+			}
+		}
+		// If xl.json is not present in read quorum disks,
+		// add .CORRUPTION prefix to the current object.
+		if len(xl.getDisks())-notFoundXLJSON < readQuorum {
+			rename(ctx, xl.getDisks(), bucket, object, bucket, object+xlCorruptedSuffix, true, len(xl.getDisks())/2+1, []error{errFileNotFound})
+			// Return err file not found since we renamed now the corrupted object
+			return objInfo, errFileNotFound
+		}
+	} else {
+		// If this is a corrupted object, change read quorum to N/2 disks
+		// so it will be visible to users, so they can delete it.
+		readQuorum = len(xl.getDisks()) / 2
```

Comment:
We can keep the read quorum similar to standard? And not n/2 ?

Reply:
> We can keep the read quorum similar to standard? And not n/2 ?

We can't actually because otherwise the object won't be visible, the object is already corrupted anyway